icicle-compiler/main/icicle.hs

```diff
@@ -269,7 +279,19 @@ runQuery f msrc query = do
   let compSecs = realToFrac (queryCompilationTime query) :: Double
   liftIO (printf "icicle: compilation time = %.2fs\n" compSecs)
 
-  liftIO (maybe (return ()) (flip writeUtf8 (querySource query)) msrc)
+  case msrc of
+    Nothing ->
+      pure ()
+    Just src ->
+      writeUtf8 src (querySource query)
+
+  case sfOutputSchema (queryFleet query) of
+    Nothing ->
+      pure ()
```

Comment:
when will this be `Nothing` - should it be an error? if you specify `--output-schema blah` and there's no schema, maybe it should warn